File: FSharpDemoPresentation_C_MVC/FSharpDemoPresentation/FSharpDemoPresentation/Controllers/HomeController.cs
using FSharpDemoPresentation.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using System.Data;

namespace FSharpDemoPresentation.Controllers
{
    public class HomeController : Controller
    {
        public ViewResult Index()
        {
            return View(ChaptersPresentationModel.DbChapters);
        }

        public ViewResult DemoContent(string title, string category)
        {
            ViewBag.Title = title;
            ViewBag.ContentPartialView = category;
            return View(ChaptersPresentationModel.DbChapters);
        }

        #region Dynamic view (use partial)
        public ViewResult Presentation()
        {
            var chapters = ChaptersPresentationModel.DbChapters;
            //var rslt = chapters.AsEnumerable<ChapterPresentation>()
            //    .Join((new List<string>() { "Introduction", "Conclusion" }).AsEnumerable<string>(),
            //            chp => chp.Title, sec => sec, (chp, tit) => chp)
            //    .Select(chp => chp)
            //    .ToList<ChapterPresentation>();
            //return View(rslt);
            return View(chapters);
        }

        public PartialViewResult GetCategory(string title, string category)
        {
            ViewBag.Title = title;
            var selectedCategory = ChaptersPresentationModel.DbChapters.Where(chap => chap.Title.Equals(title)).ToList();
            return PartialView(category, selectedCategory);
        }
        #endregion

        #region Contact & About
        public ActionResult About()
        {
            ViewBag.Message = "This application is just a presentation of F# programming.";

            return View(ChaptersPresentationModel.DbChapters);
        }

        public ActionResult Contact()
        {
            ViewBag.Message = "Pierre Contri";

            return View(ChaptersPresentationModel.DbChapters);
        }
        #endregion
    }
}

File: FSharpDemoPresentation_FsMVC5/FSharpDemoPresentation_FsMVC5/FSharpDemoPresentation_FsMVC5/Controllers/HomeController.fs
namespace FSharpDemoPresentation_FsMVC5.Controllers

open System
open System.Collections.Generic
open System.Linq
open System.Web
open System.Web.Mvc
open System.Web.Mvc.Ajax

type HomeController() =
    inherit Controller()
    member this.Index () = this.View() :> ActionResult

    member this.IndexCategory (title: string) (category: string) = 
        this.ViewData.Add("Title", title)
        this.ViewData.Add("ContentPartialView", "IndexSubCategories/" + category)
        this.View("DemoContent") :> ActionResult

    member this.GetCategory () =
        let title = this.Request.Params.GetValues("title").FirstOrDefault()
        let partialView = "IndexSubCategories/" + this.Request.Params.GetValues("category").FirstOrDefault()
        this.PartialView(partialView) :> ActionResult

    member this.Presentation () = this.View() :> ActionResult
    member this.Contact () = this.View() :> ActionResult
    member this.About () = 
        this.ViewData.Add("Message", "This application is just a presentation of F# programming.")
        this.View() :> ActionResult
